### PR TITLE
clientv3/integration: add TestKVDelete*

### PR DESCRIPTION
**Pull request by gyuho**
/cc @xiang90 @heyitsanthony 

Thanks!

### CODE REVIEW

File: clientv3/integration/kv_test.go [hunk 1]
```diff
@@ -179,3 +179,104 @@ func TestKVRange(t *testing.T) {
...
+
+	for i, tt := range tests {
+		dresp, err := kv.DeleteRange(tt.key, tt.end)
+		if err != nil {
+			t.Fatalf("#%d: couldn't delete range (%v)", i, err)
+		}
+		if dresp.Header.Revision != tt.delRev {
+			t.Fatalf("#%d: dresp.Header.Revision got %d, want %d", i, dresp.Header.Revision, tt.delRev)
+		}
+		grespKey, err := kv.Get(tt.key, 0)
```
Comment (xiang90) [inline clientv3/integration/kv_test.go]: we do not have to test get here. we should assume the server side do the current thing. but it does not hurt either.

Comment (xiang90) [inline clientv3/integration/kv_test.go]: probably we should just do a range and expect nothing in the range.

File: clientv3/integration/kv_test.go [hunk 2]
```diff
@@ -179,3 +179,104 @@ func TestKVRange(t *testing.T) {
...
+		t.Fatalf("failed to create lease %v", err)
+	}
+
+	tests := []struct {
+		key, val       string
+		leaseID        lease.LeaseID
+		putRev, delRev int64
+	}{
+		{"foo", "bar", lease.NoLease, 2, 3},
+		{"hello", "world", lease.LeaseID(resp.ID), 4, 5},
```
Comment (xiang90) [inline clientv3/integration/kv_test.go]: this test case is irrelevant i think. leaseID is not an input of delete. so we probably only need to test one removal case. and it does not need to be table driven.

File: clientv3/integration/kv_test.go (hunk 1 above)
Comment (gyuho) [inline clientv3/integration/kv_test.go]: Ok will fix. Thanks.

File: clientv3/integration/kv_test.go (hunk 2 above)
Comment (gyuho) [inline clientv3/integration/kv_test.go]: makes sense. Will fix. thanks.

**Comment by gyuho** [comment]
@xiang90 Just addressed. PTAL.

Thanks,

**Comment by xiang90** [comment]
LGTM

**Comment by gyuho** [comment]
Thanks!

**Pull request merged**